pkg/v1/providers/infrastructure-vsphere/v1.1.0/ytt/overlay.yaml

```diff
@@ -316,6 +311,15 @@ spec:
     #@overlay/append
     - sed -i '/listen-client-urls/ s/$/,https:\/\/127.0.0.1:2379/' /etc/kubernetes/manifests/etcd.yaml
     #@ end
+    #! When using kube-vip as the control plane endpoint provider with IPv6 as
+    #! the primary IP family, set --node-ip on kubelet so that host network pods
+    #! do not get the kube-vip address as their pod IP.
+    #! See: https://github.com/vmware-tanzu/tanzu-framework/issues/2098
+    #@ if not data.values.AVI_CONTROL_PLANE_HA_PROVIDER and data.values.TKG_IP_FAMILY in ["ipv6", "ipv6,ipv4"]:
+    preKubeadmCommands:
+    #@overlay/append
+    - echo "KUBELET_EXTRA_ARGS=--node-ip=$(ip -6 -json addr show dev eth0 scope global | jq -r .[0].addr_info[0].local)" >> /etc/sysconfig/kubelet
```

Comment:
Will jq be present in all cases? Do Photon images contain jq?

Reply:
Good call out that we're introducing a dependency here on `jq` that may otherwise not be required. 

I tested this on Ubuntu using an official ova and on Photon using a custom build and felt it was safe to assume, but I'll leave this thread open to see if others have thoughts on this.

Reply:
reached out to some folks more familiar with image-builder and was told:

> All node images built via image-builder install jq currently - https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/ansible/roles/node/defaults/main.yml#L21
Introducing a jq dependency should be okay.